oauth2.auth.encode_none() changes the body size but leaves content-length header set to the old size

**Describe the bug**

When using the HTTPX AsyncOAuth2Client with password grant, the httpx_client OAuth2ClientAuth.auth_flow() method modifies the body by adding the client_id, but it does not update the Content-Length header which has already been calculated.   This causes an exception in the httpx h11 processing when it deletes more characters from the stream buffer than the content-length has specified.

**Error Stacks**

```
.virtualenvs/traffica_stc/lib/python3.7/site-packages/authlib/integrations/httpx_client/oauth2_client.py:109: in _fetch_token
    auth=auth, **kwargs)
.virtualenvs/traffica_stc/lib/python3.7/site-packages/httpx/client.py:1316: in post
    timeout=timeout,
.virtualenvs/traffica_stc/lib/python3.7/site-packages/authlib/integrations/httpx_client/oauth2_client.py:89: in request
    method, url, auth=auth, **kwargs)
.virtualenvs/traffica_stc/lib/python3.7/site-packages/httpx/client.py:1097: in request
    request, auth=auth, allow_redirects=allow_redirects, timeout=timeout,
.virtualenvs/traffica_stc/lib/python3.7/site-packages/httpx/client.py:1118: in send
    request, auth=auth, timeout=timeout, allow_redirects=allow_redirects,
.virtualenvs/traffica_stc/lib/python3.7/site-packages/httpx/client.py:1148: in send_handling_redirects
    request, auth=auth, timeout=timeout, history=history
.virtualenvs/traffica_stc/lib/python3.7/site-packages/httpx/client.py:1184: in send_handling_auth
    response = await self.send_single_request(request, timeout)
.virtualenvs/traffica_stc/lib/python3.7/site-packages/httpx/client.py:1208: in send_single_request
    response = await dispatcher.send(request, timeout=timeout)
.virtualenvs/traffica_stc/lib/python3.7/site-packages/httpx/dispatch/connection_pool.py:157: in send
    raise exc
.virtualenvs/traffica_stc/lib/python3.7/site-packages/httpx/dispatch/connection_pool.py:153: in send
    response = await connection.send(request, timeout=timeout)
.virtualenvs/traffica_stc/lib/python3.7/site-packages/httpx/dispatch/connection.py:44: in send
    return await self.connection.send(request, timeout=timeout)
.virtualenvs/traffica_stc/lib/python3.7/site-packages/httpx/dispatch/http11.py:51: in send
    await self._send_request_body(request, timeout)
.virtualenvs/traffica_stc/lib/python3.7/site-packages/httpx/dispatch/http11.py:101: in _send_request_body
    await self._send_event(event, timeout)
.virtualenvs/traffica_stc/lib/python3.7/site-packages/httpx/dispatch/http11.py:117: in _send_event
    bytes_to_send = self.h11_state.send(event)
.virtualenvs/traffica_stc/lib/python3.7/site-packages/h11/_connection.py:464: in send
    data_list = self.send_with_data_passthrough(event)
.virtualenvs/traffica_stc/lib/python3.7/site-packages/h11/_connection.py:498: in send_with_data_passthrough
    writer(event, data_list.append)
.virtualenvs/traffica_stc/lib/python3.7/site-packages/h11/_writers.py:69: in __call__
    self.send_data(event.data, write)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <h11._writers.ContentLengthWriter object at 0x7fdbaa265a58>
data = b'grant_type=password&username=foo&password=barbar%23123&client_id=foo-backend'
write = <built-in method append of list object at 0x7fdbaa25ba88>

    def send_data(self, data, write):
        self._length -= len(data)
        if self._length < 0:
            raise LocalProtocolError(
>               "Too much data for declared Content-Length")
E           h11._util.LocalProtocolError: Too much data for declared Content-Length

.virtualenvs/traffica_stc/lib/python3.7/site-packages/h11/_writers.py:89: LocalProtocolError
---------------------------- Captured log teardown -----------------------------
```

**To Reproduce**

```
import pytest
from authlib.integrations.httpx_client import AsyncOAuth2Client


@pytest.mark.asyncio
async def test_keycloak():
    client = AsyncOAuth2Client(client_id="foo-backend",
                               client_secret=None,
                               username="foo", password="barbar#123",
                               token_endpoint="https://keycloak/auth/realms/myrealm/protocol/openid-connect/token",
                               verify=False, trust_env=False)
    client.token = await client.fetch_token(url="https://keycloak/auth/realms/myrealm/protocol/openid-connect/token", username="foo",
                                            password="barbar#123")
    print(client.token)
```
**Expected behavior**

The token should be fetched from the server

**Environment:**

 - OS: CentOS Linux
 - Python Version: 3.7
 - Authlib Version: 0.14

**Additional context**

This change in oauth2/auth.py fixes the problem:

```
def encode_none(client, method, uri, headers, body):
    if method == 'GET':
        uri = add_params_to_uri(uri, [('client_id', client.client_id)])
        return uri, headers, body
    body = add_params_to_qs(body, [('client_id', client.client_id)])
    # Update Content-Length header
    headers['Content-Length'] = str(len(body))
    return uri, headers, body
```

--- a/authlib/oauth2/auth.py
+++ b/authlib/oauth2/auth.py
@@ -17,6 +17,8 @@ def encode_client_secret_post(client, method, uri, headers, body):
         ('client_id', client.client_id),
         ('client_secret', client.client_secret or '')
     ])
+    if 'Content-Length' in headers:
+        headers['Content-Length'] = str(len(body))
     return uri, headers, body
 
 
@@ -25,6 +27,8 @@ def encode_none(client, method, uri, headers, body):
         uri = add_params_to_uri(uri, [('client_id', client.client_id)])
         return uri, headers, body
     body = add_params_to_qs(body, [('client_id', client.client_id)])
+    if 'Content-Length' in headers:
+        headers['Content-Length'] = str(len(body))
     return uri, headers, body
 
 

